### PR TITLE
Fixing out of valid range for stress level.

### PR DESCRIPTION
**Pull request by billw2012**
One of my dwarves hit -102k stress level and started crashing DT.

### CODE REVIEW

**Comment by cvuchener** [comment]
It will still crash if wrong values are in game_data.ini. This loop:
https://github.com/Dwarf-Therapist/Dwarf-Therapist/blob/fe217cfc6b9717784389f5003993624738069e11/src/dwarf.cpp#L1873-L1877
should be fixed instead so `m_happiness` is always a valid value (use `i < DH_TOTAL_LEVELS - 1` so the last comparison is skipped). Then the "ecstatic" could be removed from game_data.ini as it is useless.

Are you using DF 0.44.09 or older? Or did you get this stress value with the newer versions?

**Comment by billw2012** [comment]
I'm using 43.05 as part of Masterwork latest. I won't be offended if you reject my PR and fix it another way, I wanted to get it working for myself mostly, just consider this a bug report!

**Comment by cvuchener** [comment]
You should use a modified game_data.ini so the happiness is displayed correctly. Add one zero in each threshold (see "Known issues" in http://www.bay12forums.com/smf/index.php?topic=168411.0).

I am leaving this PR open until I create my own, so I don't forget.

**Comment by billw2012** [comment]
I updated to what you suggest, however it includes the changes to the ranges of game_data.ini so if you do pull this make sure to exclude that!

File: .gitignore [hunk 1]
```diff
@@ -27,3 +27,87 @@ translations/qtmobility_untranslated.ts
 translations/qtmobility_*.qm
 
 *.pro.user*
+cmake.bat
```
Comment (lethosor) [inline .gitignore]: These are excessive. You should use wildcards instead.

File: .gitignore [hunk 2]
```diff
@@ -27,3 +27,87 @@ translations/qtmobility_untranslated.ts
...
+build/CMakeFiles/DwarfTherapist_automoc.dir/AutomocOldMocDefinitions.cmake
+build/CMakeFiles/DwarfTherapist_automoc.dir/AutogenInfo.cmake
+build/CMakeFiles/DwarfTherapist.dir/qrc_resources.cpp
+build/CMakeFiles/cmake.check_cache
+build/CMakeFiles/aba01c8347d736b227d0585af636fc7f/PACKAGE_force.rule
+build/CMakeFiles/aba01c8347d736b227d0585af636fc7f/INSTALL_force.rule
+build/CMakeFiles/aba01c8347d736b227d0585af636fc7f/generate.stamp.rule
+build/CMakeFiles/aba01c8347d736b227d0585af636fc7f/DwarfTherapist_automoc.rule
+build/CMakeFiles/3.5.2/CompilerIdCXX/CompilerIdCXX.vcxproj
+build/CMakeFiles/3.5.2/CompilerIdCXX/CMakeCXXCompilerId.cpp
```
Comment (lethosor) [inline .gitignore]: In particular, these are build-specific and should not be listed individually.

**Comment by cvuchener** [comment]
Please remove your changes to .gitignore and game_data.ini. And remove your first commit now that it has become useless.

I don't mind updates to gitignore (actually I see a lot (almost all) of it is outdated so it would be welcome), but do that in a separate commit and keep it generic (see lethosor comments).

**Comment by lethosor** [comment]
Now that I'm looking on my computer, there actually isn't a `build` folder tracked in this repo at all, so just ignore `build/` (or preferably `build*/`, if you want to allow people to be flexible and/or have multiple build folders).

I'm not sure what `cmake.bat` is - it's certainly nothing I've ever seen created by CMake, and I have no idea why it's outside the build folder.

**Comment by billw2012** [comment]
Yeah none of those changes were intended to be in the PR, I just checked in and they got automatically added to it. I have no idea how PRs work, particularly when you receive them. I kind of assumed you can just select what files you actually want to accept from the PR. But I'm not doing anything else, this is literally a 2 char change (as @cvuchener said this is just left here for a reminder), just do it manually!

**Comment by lethosor** [comment]
Pull requests automatically include any commits that you make to the branch in your repo that you're requesting to be merged. You selected your repo's `master` branch when you opened this PR (or maybe there wasn't another option), so any commits to that branch will be included here. A better way to do it (although perhaps too late) is to make a separate branch for just the changes you want merged, and push the changes to that branch.

**Pull request closed without merging**